codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectEncoder.java

```diff
@@ -307,22 +307,42 @@ private void encodeFullHttpMessage(ChannelHandlerContext ctx, Object o, List<Obj
 
             final H m = (H) o;
 
-            final ByteBuf buf = ctx.alloc().buffer((int) headersEncodedSizeAccumulator);
-
-            encodeInitialLine(buf, m);
-
             final int state = isContentAlwaysEmpty(m) ? ST_CONTENT_ALWAYS_EMPTY :
                     HttpUtil.isTransferEncodingChunked(m) ? ST_CONTENT_CHUNK : ST_CONTENT_NON_CHUNK;
 
+            ByteBuf content = msg.content();
+            // try to save adding the content as an additional buffer to the out list by merging it into the headers
+            final boolean copyContent = content.readableBytes() > 0 &&
```

Comment:
needCopyContent？

Reply:
It should be as good as it is, but I'm not a native eng speaker, so i am not sure!

Reply:
I change it again to reflect that we just account of the content bytes instead, hoping the copy would happen